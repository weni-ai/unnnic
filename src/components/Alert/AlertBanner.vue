<template>
    <section :class="['banner-alert', {
        'banner-alert--banner-danger': type === 'danger',
        'banner-alert--banner-warning': type === 'warning',
      }]">
        <header v-show="text" class="banner-alert__container-text">
          <unnnic-icon class="banner-alert__textIcon" v-show="isShowTextIcon(type)" :icon="getIconType(type)" size="sm" scheme="neutral-white" />
          <p class="text" >{{ text }}</p>
          <a v-if="linkHref" class="banner-alert__link" :href="linkHref" :target="linkTarget">
            {{ linkText }}
          </a>
        </header>
  
        <div
          v-show="showCloseButton"
          class="banner-alert__close"
          @click="emitClose"
        >
          <unnnic-icon icon="close" size="sm" scheme="neutral-white" />
        </div>
      </section>
  </template>
  
  <script>
  import UnnnicIcon from '../Icon.vue';
  
  export default {
    components: {
      UnnnicIcon,
    },
    methods: {
      getIconType(type) {
        if (type === 'danger') return 'block';
        if (!type.trim()) return 'info'; //check if type is empty or whitespace
        return type;
      },
      isShowTextIcon(type) {
        return ['danger', 'warning', 'info', ''].includes(type)
      },
      emitClose() {
        this.onClose();

        this.$emit('close');
      }
    },
    props: {
      text: {
        type: String,
        default: null,
      }, 
      onClose: {
        type: Function,
        default: () => {}
      },
      showCloseButton: {
        type: Boolean,
        default: false
      },
      linkHref: {
        type: String,
        default: '',
      },
      linkTarget: {
        type: String,
        default: '_blank',
      },
      linkText: {
        type: String,
        default: 'Learn more',
      },
      type: {
        type: String,
        default: 'info',
      },
    }
  };
  </script>
  
  <style lang="scss" scoped>
  @import '../../assets/scss/unnnic.scss';
  
  .banner-alert {
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: space-between;
    margin-bottom: $unnnic-spacing-sm;
  
    color: $unnnic-color-neutral-white;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    font-weight: $unnnic-font-weight-regular;
    background-color: $unnnic-color-aux-blue-500;

    &__container-text {
      display: flex;
      justify-content: center;
      align-items: center;
      padding: $unnnic-spacing-xs $unnnic-spacing-sm;
      flex-grow: 1;

      .text {
        margin: 0 $unnnic-spacing-nano 0 0;
      }
    }

    &__textIcon {
      margin-right: $unnnic-spacing-xs;
    }
  
    &__link {
      color: inherit;
      display: block;
      font-weight: $unnnic-font-weight-bold;
      text-decoration: underline;
      text-underline-position: under;
    }
  
    &__close {
      display: flex;
      padding: 11px $unnnic-spacing-sm;
      cursor: pointer;
      user-select: none;
      border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-blue-300;
      &:hover {
          background-color: $unnnic-color-aux-blue-300;
        }
    }
  
    &--banner-warning {
      background-color: $unnnic-color-aux-orange-500;
  
      .banner-alert__close {
        border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-orange-300;
  
        &:hover {
          background-color: $unnnic-color-aux-orange-300;
        }
      }
    }
  
    &--banner-danger {
      background-color: $unnnic-color-aux-red-500;
  
      .banner-alert__close {
        border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-red-300;
  
        &:hover {
          background-color: $unnnic-color-aux-red-300;
        }
      }
    }
  }
  </style>
  