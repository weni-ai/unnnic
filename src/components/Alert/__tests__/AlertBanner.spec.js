import { mount } from '@vue/test-utils';
import { describe, test, expect, beforeEach, afterEach, vi } from 'vitest';
import AlertBanner from '@/components/Alert/AlertBanner.vue';

describe('AlertBanner.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(AlertBanner, {
      props: {
        text: 'This is a banner alert',
        type: 'danger',
        showCloseButton: true,
        linkHref: 'https://example.com',
        linkText: 'Learn more',
        linkTarget: '_self',
        onClose: vi.fn(),
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  test('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('renders the correct text and link', () => {
    const textElement = wrapper.find('.text');
    const linkElement = wrapper.find('.banner-alert__link');

    expect(textElement.text()).toBe('This is a banner alert');
    expect(linkElement.text()).toBe('Learn more');
    expect(linkElement.attributes('href')).toBe('https://example.com');
    expect(linkElement.attributes('target')).toBe('_self');
  });

  test('applies the correct class based on the type', () => {
    expect(wrapper.classes()).toContain('banner-alert--banner-danger');
  });

  test('renders the correct icon based on the type', () => {
    const icon = wrapper.findComponent('[data-test="unnnic-icon"]');
    expect(icon.exists()).toBe(true);
    expect(icon.props('icon')).toBe('block');
    expect(icon.props('size')).toBe('sm');
    expect(icon.props('scheme')).toBe('neutral-white');
  });

  test('does not render the icon when type is not danger, warning, or info', async () => {
    await wrapper.setProps({ type: 'success' });
    const icon = wrapper.findComponent('[data-test="unnnic-icon"]');
    expect(icon.element.style.display).toBe('none');
  });

  test('calls emitClose and emits close event when close button is clicked', async () => {
    const closeButton = wrapper.find('.banner-alert__close');
    await closeButton.trigger('click');

    expect(wrapper.emitted()).toHaveProperty('close');
    expect(wrapper.props().onClose).toHaveBeenCalled();
  });

  test('does not render close button if showCloseButton is false', async () => {
    await wrapper.setProps({ showCloseButton: false });
    const closeButton = wrapper.find('.banner-alert__close');
    expect(closeButton.element.style.display).toBe('none');
  });

  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('getIconType function returns the correct icon based on type', () => {
    expect(wrapper.vm.getIconType('danger')).toBe('block');
    expect(wrapper.vm.getIconType('info')).toBe('info');
    expect(wrapper.vm.getIconType('')).toBe('info');
  });

  test('isShowTextIcon function returns true for danger, warning, and info types', () => {
    expect(wrapper.vm.isShowTextIcon('danger')).toBe(true);
    expect(wrapper.vm.isShowTextIcon('warning')).toBe(true);
    expect(wrapper.vm.isShowTextIcon('info')).toBe(true);
    expect(wrapper.vm.isShowTextIcon('')).toBe(true);
    expect(wrapper.vm.isShowTextIcon('success')).toBe(false);
  });
});
