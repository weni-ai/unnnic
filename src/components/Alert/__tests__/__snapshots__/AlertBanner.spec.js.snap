// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertBanner.vue > matches the snapshot 1`] = `
"<section data-v-e8e4d051="" class="banner-alert banner-alert--banner-danger">
  <header data-v-e8e4d051="" class="banner-alert__container-text"><span data-v-26446d8e="" data-v-e8e4d051="" class="material-symbols-rounded unnnic-icon-scheme--neutral-white unnnic-icon-size--sm banner-alert__textIcon" data-testid="material-icon" translate="no" data-test="unnnic-icon">block</span>
    <p data-v-e8e4d051="" class="text">This is a banner alert</p><a data-v-e8e4d051="" class="banner-alert__link" href="https://example.com" target="_self">Learn more</a>
  </header>
  <div data-v-e8e4d051="" class="banner-alert__close"><span data-v-26446d8e="" data-v-e8e4d051="" class="material-symbols-rounded unnnic-icon-scheme--neutral-white unnnic-icon-size--sm" data-testid="material-icon" translate="no" data-test="unnnic-close-icon">close</span></div>
</section>"
`;
